### PR TITLE
fix: cbr profile defaults to all endpoints

### PR DESCRIPTION
**Pull request by Aayush-Abhyarthi**
### Description

- Support for all endpoints
- Cardinality of zone creation
- Continuous test failures

https://github.com/terraform-ibm-modules/terraform-ibm-cbr/issues/253
https://github.com/terraform-ibm-modules/terraform-ibm-cbr/issues/259

#### Changes that affect the core Terraform module or submodules
- [x] Bug fix
- [ ] New feature
- [ ] Dependency update

#### Changes that don't affect the core Terraform module or submodules
- [x] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other

#### Release required?
Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning).

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

If a release is required, replace this text with information that users need to know about the release. Write the release notes to help users understand the changes, and include information about how to update from the previous version.

Your notes help the merger write the commit message for the PR that is published in the release notes for the module.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

**Comment by Aayush-Abhyarthi** [comment]
Tried creating two contexts - one applying to private endpoint and other to public but the pipeline is failing  and throwing the error "expected contexts not found". Seems like it allows only one context per zone.
A similar  [PR](https://github.com/terraform-ibm-modules/terraform-ibm-cbr/pull/254) concerning this issue throws the same error. 
Any thoughts regarding this?
@vburckhardt @jojustin

**Comment by Ak-sky** [comment]
> Tried creating two contexts - one applying to private endpoint and other to public but the pipeline is failing and throwing the error "expected contexts not found". Seems like it allows only one context per zone. A similar [PR](https://github.com/terraform-ibm-modules/terraform-ibm-cbr/pull/254) concerning this issue throws the same error. Any thoughts regarding this? @vburckhardt @jojustin

If the endpoints has to be set to both public and private, attribute value need to be set as  `"value": "public,private"`.

File: cbr-service-profile/variables.tf [hunk 1]
```diff
@@ -60,3 +60,15 @@ variable "target_service_details" {
     error_message = "Provide a valid target service name that is supported by context-based restrictions"
   }
 }
+
+variable "endpoints" {
+  type        = list(string)
+  description = "List specific endpoint types for target services, valid values for endpoints are 'public', 'private' or 'direct'"
+  default     = []
```
Comment (vburckhardt) [inline cbr-service-profile/variables.tf:67]: I think we should default to ["private"] here.

Comment (Ak-sky) [reply cbr-service-profile/variables.tf]: done

**Review by vburckhardt — APPROVED** [review]
👍

**Pull request merged**